### ISSUE TITLE
External services defined through a `ServiceEntry` become accessible to all applications in the Mesh. There should be a way restrict accessibility of those external services

### ISSUE THREAD

**Issue by yskopets** [istio/istio#9477]
**Describe the feature request**

There should be a way to restrict accessibility of external services defined through a `ServiceEntry`.

E.g.,
A. `ServiceEntry` should come in 2 flavours: 1) "really" namespace-scoped and 2) cluster-scoped
B. `RBAC Policies` should work for outgoing requests

**Describe alternatives you've considered**

We tried to solve the problem by using `sourceLabels` constraint in a `VirtualService` (so that different applications could see different routing rules).

Unfortunately, the solution if full of tricks as we, essentially, go against the Istio's "global" model.

**Additional context**
* We're using Istio in a multi-tenant environment (where a single Kubernetes cluster is shared by multiple teams and each of them is expected to be autonomous)
* Every tenant (team) expects to have full control over its own application (Namespace)
* However, as of v1.0.x, `Istio` only supports a single "global" configuration that affects every tenant and can be modified/corrupted by every tenant

**Comment by eduardobaitello** [comment]
Maybe related to #8462

**Comment by esnible** [comment]
Have you tried defining your ServiceEntry with the sourceLabels feature of https://istio.io/docs/reference/config/istio.networking.v1alpha3/#HTTPMatchRequest ?  You should be able to specify that only pods with a particular set of labels may use the ServiceEntry.
